aws/data_source_aws_dx_private_virtual_interface.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsDxPrivateVirtualInterface() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDxPrivateVirtualInterfaceRead,
+
+		Schema: map[string]*schema.Schema{
+			"virtual_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"connection_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpn_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dx_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vlan": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"bgp_asn": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"bgp_auth_key": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
`Optional: true` should be removed from this attribute since it cannot be used to determine the VIF 👍

Reply:
Oops! Fixed